### ISSUE TITLE
JsonArray: The node must be of type 'JsonObject'.

### ISSUE THREAD

**Issue by hell-racer** [Havunen/SystemTextJsonPatch#15]
Consider the following minimalistic test:

```
[Fact]
public void MyTestMethod()
{
    var options = new JsonSerializerOptions() { Converters = { new SystemTextJsonPatch.Converters.JsonPatchDocumentConverterFactory() } };

    JsonArray node = JsonSerializer.Deserialize<JsonArray>(@"[ 123 ]");
    SystemTextJsonPatch.JsonPatchDocument patch = JsonSerializer.Deserialize<SystemTextJsonPatch.JsonPatchDocument>(@"[ { ""op"": ""replace"", ""path"": ""/0"", ""value"": 456 } ]", options);
    patch.ApplyTo(node);
}
```

I'm trying to replace first (and single) element of the array. The following exception is thrown:
`System.InvalidOperationException: 'The node must be of type 'JsonObject'.'`

The same code works with Newtonsoft:

```
[Fact]
public void MyTestMethod()
{
    JArray nnode = JsonConvert.DeserializeObject<JArray>(@"[ 123 ]");
    Microsoft.AspNetCore.JsonPatch.JsonPatchDocument patch = JsonConvert.DeserializeObject<Microsoft.AspNetCore.JsonPatch.JsonPatchDocument>(@"[ { ""op"": ""replace"", ""path"": ""/0"", ""value"": 456 } ]");
    patch.ApplyTo(nnode);
}
```

**Comment by Havunen** [comment]
Hi, thanks for reporting this issue.
This sounds like a bug. I'll check

**Comment by Havunen** [comment]
This issue is now fixed and version 1.2.0 has been published in Nuget
